src/Raven.Studio/typescript/viewmodels/database/indexes/deleteIndexesConfirm.ts

```diff
@@ -3,33 +3,75 @@ import dialog = require("plugins/dialog");
 import database = require("models/resources/database");
 import dialogViewModelBase = require("viewmodels/dialogViewModelBase");
 import messagePublisher = require("common/messagePublisher");
+import index = require("models/database/index/index");
+import indexDefinition = require("models/database/index/indexDefinition");
 
-class deleteIndexesConfirm extends dialogViewModelBase {
+class indexInfoForDelete {
+    indexName: string;
+    reduceOutputCollection: string;
+    referenceCollection: string;
 
-    deleteTask = $.Deferred<boolean>();
-    title: string;
+    readonly referenceCollectionExtension = "/References";
+    
+    static fromIndex(index: index) {
+        return new indexInfoForDelete(index.name, index.reduceOutputCollection(), !!index.reduceOutputReferencePattern());   
+    }
+    
+    static fromIndexDefinition(index: indexDefinition) {
+        return new indexInfoForDelete(index.name(), index.reduceToCollectionName(), index.patternForOutput());
+    }
+    
+    private constructor(indexName: string, reduceOutputCollection: string, hasReferenceCollection: boolean) {
+        this.indexName = indexName;
+        this.reduceOutputCollection = reduceOutputCollection;
+        this.referenceCollection = hasReferenceCollection ? reduceOutputCollection + this.referenceCollectionExtension : "";
+    }
+}
 
-    constructor(private indexNames: string[], private db: database, title?: string) {
+class deleteIndexesConfirm extends dialogViewModelBase {
+    title: string;
+    subTitleHtml: string;
+    showWarning: boolean = false;
+    deleteTask = $.Deferred<boolean>();
+    
+    indexesInfoForDelete = ko.observableArray<indexInfoForDelete>();
+    
+    constructor(private indexes: index[] | indexDefinition[], private db: database) {
         super();
 
-        if (!indexNames || indexNames.length === 0) {
+        if (!indexes || indexes.length === 0) {
             throw new Error("Indexes must not be null or empty.");
         }
 
-        this.title = title || (indexNames.length === 1 ? "Delete index?" : "Delete indexes?");
+        if (indexes[0] instanceof index) {
+            this.indexesInfoForDelete((indexes as index[]).map(x => indexInfoForDelete.fromIndex(x)));
+        } else {
+            this.indexesInfoForDelete((indexes as indexDefinition[]).map(x => indexInfoForDelete.fromIndexDefinition(x)));
+        }
+        
+        if (this.indexesInfoForDelete().length === 1) {
+            this.title = "Delete index?";
+            this.subTitleHtml = `You're deleting index:`;
+        } else {
+            this.title = "Delete indexes?";
+            this.subTitleHtml = `You're deleting <strong>${this.indexesInfoForDelete().length}</strong> indexes:`;
+        }
+
+        this.showWarning = _.some(this.indexesInfoForDelete(), x => x.reduceOutputCollection);
```

Comment:
please avoid having variables which relay on observables since it might cause issues. Either make showWarning computed or make indexesInfoForDelete array (w/o observable)

Reply:
done - made `indexesInfoForDelete` regular array